From 86fe847f4d6e06c926e98b108063685bdbf332ee Mon Sep 17 00:00:00 2001
Message-Id: <86fe847f4d6e06c926e98b108063685bdbf332ee@dist-git>
From: Erik Skultety <eskultet@redhat.com>
Date: Wed, 10 Aug 2016 15:33:12 +0200
Subject: [PATCH] virt-admin: Fix the error when an invalid URI has been
 provided

After commit 9d479dd1 fiddled with the cmdConnect's output which used to be a
bit more verbose prior to the mentioned commit, the program flow would result
in a quite confusing error if an invalid URI has been provided:

    error: Failed to connect to the admin server
    Connected to the admin server
    error: <some error>

The problem is that the commit mentioned above relied on the fact that
connect routine always succeeds which is not true.

Signed-off-by: Erik Skultety <eskultet@redhat.com>
(cherry picked from commit 4914494eb8e25ff7938a12fd089c746ec6042bbb)

https://bugzilla.redhat.com/show_bug.cgi?id=1365903
Signed-off-by: Erik Skultety <eskultet@redhat.com>
---
 tools/virt-admin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/virt-admin.c b/tools/virt-admin.c
index 2ae05da..513054b 100644
--- a/tools/virt-admin.c
+++ b/tools/virt-admin.c
@@ -351,7 +351,7 @@ cmdConnect(vshControl *ctl, const vshCmd *cmd)
     }
 
     vshAdmReconnect(ctl);
-    if (!connected)
+    if (!connected && priv->conn)
         vshPrint(ctl, "%s\n", _("Connected to the admin server"));
 
     return !!priv->conn;
-- 
2.9.2

